standard/statements.md

```diff
@@ -414,6 +417,78 @@ The scope of a local constant is the block in which the declaration occurs. It i
 
 A local constant declaration that declares multiple constants is equivalent to multiple declarations of single constants with the same type.
 
+### §local-function-declarations-new-clause Local function declarations
+
+A *local_function_declaration* declares a local function.
+
+```ANTLR
+local_function_declaration
+    : local_function_header local_function_body
+    ;
+
+local_function_header
+    : local_function_modifier* return_type identifier type_parameter_list?
+        ( formal_parameter_list? ) type_parameter_constraints_clause*
+    ;
+local_function_modifier
+    : 'async'
+    | 'unsafe'
+    ;
+
+local_function_body
+    : block
+    | '=>' expression ';'
+    | '=>' null_conditional_invocation_expression ';'
+    ;
+```
+
+Grammar note: When recognising a *local_function_body* if both the *null_conditional_invocation_expression* and *expression* alternatives are applicable then the former shall be chosen.
```

Comment:
Can we reorder the options in the grammar then? In particular it's odd for the former in the prose to be the latter in the grammar, as well as making more sense (to me anyway) as "first match wins".

Reply:
https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/classes.md#146-methods has null_conditional_invocation_expression first with the same text as is here.

But there's no example there either.

Suggest we add an example in 14.6 and refer to it from here.

Reply:
11.7.9 gives a little more motivation, but still doesn't have an expression-bodied member example.

Reply:
This discussion is captured in #621 

Edit here to add a link to §14.6.1.